src/utils/input/utils.js

```diff
@@ -44,13 +44,17 @@ export const splitSource = source => {
 	};
 };
 
-const timeoutPromise = ms =>
-	new Promise((resolve, reject) => {
-		const id = setTimeout(() => {
-			clearTimeout(id);
-			reject(new Error(`Timed out after ${ms} ms`));
-		}, ms);
-	});
+const timeoutPromise = async readFromStd => {
+	try {
+		await (() =>
+			new Promise(([reject]) => setTimeout(reject, DEFAULT_TIMEOUT)))();
```

Comment:
Shouldn't we `clearTimeout` in normal scenario?
Also, this doesn't throw error, so what are we doing for `try-catch`...? do we need `Promise` here?

Reply:
I haven't thought about how to clean it up, but intended functionality is something like this?
Maybe we can chain this function to `handleClose` part so it uses `getRawStdIn` in `getStdIn` and it just handles the output nicely?
```
const readFromStd = new Promise((resolve, reject) => {
	const rl = readline.createInterface({ input: process.stdin });
	const lines = [];
	rl
		.on('line', line => lines.push(line))
		.on('close', () => resolve(lines));

	const id = setTimeout(() => {
		clearTimeout(id);
		if (stdinIsTTY()) {
			reject(new Error(`Timed out after ${DEFAULT_TIMEOUT} ms`));
		} else {
			resolve(lines);
		}
	}, DEFAULT_TIMEOUT);
});
return readFromStd;
```